.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,46 @@
+name: Doxygen
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  CI_Doxygen:
+    name: ubuntu_debug_x64
+    runs-on: ubuntu-latest
+    container:
+      image: novelrt/novelrt-build:latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Run CI Build Script (Engine & Docs only)
+        run: scripts/cibuild.sh --configuration Debug --generate -DNOVELRT_BUILD_TESTS OFF -DNOVELRT_BUILD_SAMPLES OFF
+
+      - name: Generate Documentation
+        uses: mattnotmitt/doxygen-action@v1
+        id: generate
+        with:
+            working-directory: '$GITHUB_WORKSPACE/novelrt/artifacts/build/Debug'
+            doxyfile-path: 'doxygen/Doxyfile.doxygen'
+
+      - name: Copy Documentation
+        id: copy
+        if: ${{ success() }}
+        run: mkdir -p $GITHUB_WORKSPACE/novelrt/docs && cp -r -f -v $GITHUB_WORKSPACE/novelrt/artifacts/build/Debug/doxygen/html/* $GITHUB_WORKSPACE/novelrt/docs
+
+      - name: Get current date
+        id: date
+        run: echo "name=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
```

Comment:
We probably don't actually need the date here since the `git commit` step includes the date in the git log anyway. What would be more useful, IMO, is the commit hash that triggered the build, which you can get using `$GITHUB_SHA`

Reply:
This _should_ be resolved.